content/artifacts/artifacts-helper/plugin.js

```diff
@@ -0,0 +1,241 @@
+            >${props.planet.locationId.substring(4, 10)}</span>
+        <span style=${{ flex: 1 }}></span>
+        <span style=${{ ...Spacing }}>${PlanetType2Str[props.planetStatus]}</span>
+        </span>
+    </div>
+    `;
+}
+
+function AppUI(props) { // refreshPlanetStates, doProspectAndFind, chainBlockNumber, planets
+    //const [chainBlockNumber, setChainBlockNumber] = useState(0);
+    //const [planets, setPlanets] = useState([]);
+    const itemsHTML = props.planets.map((p, i) => {
+        return html`<${PlanetItem} id=${i} planet=${p[0]} planetStatus=${p[1]} />`;
+    });
+    const nums = props.planets.reduce((acc, p) => {
+        acc[p[1]] += 1;
+        return acc;
+    }, PlanetType2Str.map(() => 0));
+    const numsHTML = PlanetType2Str.map((p, i) => html`<span style=${{ ...Spacing }}>${p}<br/>${nums[i]}</span>`);
+    const warningHTML = props.chainBlockNumber == 0 ? html`<div>Warning! No current block height info.
+        Planet status might be wrong. Please retry after several seconds. (~8s)</div>` : '';
+
+    return html`
+    <div>
+        <div style=${{ display: 'flex' }}>
+            <button style=${{ ...VerticalSpacing, ...Spacing }} onClick=${() => props.refreshPlanetStates()}>Refresh</button>
+            <button style=${{ ...VerticalSpacing, ...Spacing }} onClick=${() => props.doProspectAndFind()}>Auto P+F</button>
+            <span style=${{ flex: 1 }}></span>
+            <span style=${Spacing}>Cur Blk: ${props.chainBlockNumber < 1 ? "Loading" : props.chainBlockNumber}</span>
+        </div>
+        <div style=${{ display: 'flex' }}>
+            ${numsHTML}
+        </div>
+        ${warningHTML}
+        <div style=${{
+            maxHeight: "300px", overflowX: "hidden", overflowY: "scroll",
+            display: 'grid', gridTemplateColumns: 'repeat(3,1fr)', border: '1px solid gray'
+        }}
+            otherstyle="grid-template-columns:repeat(auto-fill, minmax(4em, max-content));">
+            ${itemsHTML}
+        </div>
+    </div>
+    `;
+}
+
+
+class Plugin {
+    //container: any; // Plugin container, used in render()
+    //pluginHTMLNode: any; // Root node of the plugin panel
+
+    //_chainBlockNumber_Subscriber: any;
+    //chainBlockNumber: number;
+    //chainBlockNumberplanets: any[];
+    container;
+    pluginHTMLNode;
+    planets;
+    chainBlockNumber;
+    _chainBlockNumber_Subscriber;
+
+    constructor() {
+        this._chainBlockNumber_Subscriber = ui.getEthConnection().blockNumber$.subscribe(x => {
+            let t = this.chainBlockNumber;
+            this.chainBlockNumber = x;
+            if (t < 1) {
+                this.refreshPlanetStates();
+            }
+        });
+        this.container = null;
+        this.pluginHTMLNode = null;
+        this.chainBlockNumber = 0;
+        this.planets = [];
+    }
+
+    destroy() {
+        this._chainBlockNumber_Subscriber.unsubscribe();
+        render(null, this.container, this.pluginHTMLNode);
```

Comment:
This changed in preact
```suggestion
        render(null, this.container);
```

Reply:
According to document of v10, the `replaceNode` argument still exists and not marked as deprecated?